app/stac_api/signals.py

```diff
@@ -1,13 +1,34 @@
 import logging
 
+from django.db.models import ProtectedError
 from django.db.models.signals import pre_delete
 from django.dispatch import receiver
 
 from stac_api.models import Asset
+from stac_api.models import AssetUpload
 
 logger = logging.getLogger(__name__)
 
 
+@receiver(pre_delete, sender=AssetUpload)
+def check_on_going_upload(sender, instance, **kwargs):
+    if instance.status == AssetUpload.Status.IN_PROGRESS:
```

Comment:
Would it make sense to force-abort the ongoing upload and perform the deletion afterwards?

Reply:
I would not because it is usually not a use case and it increase complexity.